### PR TITLE
Adds content-visibility auto to below the fold loader

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
I was following [this guide](https://web.dev/content-visibility/) on how to improve the browser's rendering  pipeline. 

Since we are already following the above/below the fold split of rendering I think we don't run into much trouble in here, however I thought it was good to give it a try. 

After this PR, I've got an average reduction of 50ms in rendering on btglobal

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
[btglobal](https://github.com/vtex-sites/btglobal.store/pull/338)
[marinbrasil](https://github.com/vtex-sites/marinbrasil.store/pull/425)
[storecomponents](https://github.com/vtex-sites/storecomponents.store/pull/686)

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

**Review by FabianaFerreira — APPROVED** [review]
Just don't forget to rebase the branch `next` with master after merging this PR to make sure that we can release a new alpha version with every change, in order not to break C4

**Review by Gmantiqueira — APPROVED** [review]
Nice!

File: packages/gatsby-theme-store/src/components/HomePage/BelowTheFoldPreview.tsx [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-    <Spinner />
-  </Center>
+  <div
+    style={
+      ({
+        contentVisibility: 'auto',
+        containIntrinsicSize: '500px',
+      } as unknown) as CSSProperties
+    }
+  >
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-store/src/components/HomePage/BelowTheFoldPreview.tsx:13]: shouldn't that be a variant?

File: packages/gatsby-theme-store/src/components/ProductPage/BelowTheFoldPreview.tsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
-  <Center height="500px">
-    <Spinner />
-  </Center>
+  <div
+    style={
+      ({
+        contentVisibility: 'auto',
+        containIntrinsicSize: '500px',
+      } as unknown) as CSSProperties
+    }
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-store/src/components/ProductPage/BelowTheFoldPreview.tsx:12]: too

File: packages/gatsby-theme-store/src/components/SearchPage/BelowTheFoldPreview.tsx [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
-    <Spinner />
-  </Center>
+  <div
+    style={
+      ({
+        contentVisibility: 'auto',
+        containIntrinsicSize: '500px',
+      } as unknown) as CSSProperties
+    }
+  >
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-store/src/components/SearchPage/BelowTheFoldPreview.tsx:13]: 👀

File: packages/gatsby-theme-store/src/components/HomePage/BelowTheFoldPreview.tsx (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-theme-store/src/components/HomePage/BelowTheFoldPreview.tsx:13]: Hmm, so. I'm not sure. 

I was thinking this weekend and I think we should remove all theme-ui's dependencies from theme-store and leave it as a standalone package. This way we can use any styling with it. What do you think ?

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**